src/MappingOperation/Implementations/MapToMultiple.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace AutoMapperPlus\MappingOperation\Implementations;
+
+use AutoMapperPlus\Exception\UnregisteredMappingException;
+use AutoMapperPlus\MappingOperation\ContextAwareOperation;
+use AutoMapperPlus\MappingOperation\ContextAwareTrait;
+use AutoMapperPlus\MappingOperation\DefaultMappingOperation;
+use AutoMapperPlus\MappingOperation\MapperAwareOperation;
+use AutoMapperPlus\MappingOperation\MapperAwareTrait;
+use Traversable;
+use function is_array;
+
+/**
+ * Class MapTo.
+ * Allows a property to be mapped itself.
+ * @package AutoMapperPlus\MappingOperation\Implementations
+ */
+class MapToMultiple extends DefaultMappingOperation implements
+    MapperAwareOperation,
+    ContextAwareOperation
+{
+    use MapperAwareTrait;
+    use ContextAwareTrait;
+
+    /**
+     * @var string
+     */
+    private $destinationClassList;
+
+    /**
+     * @var array
+     */
+    private $ownContext = [];
+
+    /**
+     * MapTo constructor.
+     * @param string[] $destinationClassList
+     * @param array $context
+     *   Indicates whether or not an array as source value should be treated as
+     *   a collection of elements, or as an array representing an object.
+     * @param array
+     *   $context Optional context that will be merged with the parent's
+     *   context.
+     */
+    public function __construct(
+        array $destinationClassList,
+        array $context = []
+    ) {
+        $this->destinationClassList = $destinationClassList;
+        $this->ownContext = $context;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getDestinationClassList(): array
+    {
+        return $this->destinationClassList;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function getSourceValue($source, string $propertyName)
+    {
+        $value = $this->propertyReader->getProperty(
+            $source,
+            $this->getSourcePropertyName($propertyName)
+        );
+
+        $context = array_merge($this->context, $this->ownContext);
+        $returnValue = null;
+        $mappingFailed = false;
+        foreach ($this->destinationClassList as $destinationClass) {
+            if ($this->isCollection($value) === false) {
+                try {
+                    $returnValue = $this->mapper->map($value, $destinationClass, $context);
+                    $mappingFailed = false;
+                    break;
+                } catch (UnregisteredMappingException $exception) {
+                    $mappingFailed = true;
+                }
+            } else {
+                foreach ($value as $item) {
```

Comment:
It is not guaranteed the source property is iterable - if the user makes a mistake in their mapping, this would fatal out. It's better to explicitly check for `is_iterable` and throw a specific exception if it is not.

Reply:
Disregard this - I'm getting a bit too tired it seems :slightly_smiling_face: